plugins/scaffolder/src/components/TemplateExtensionsPage/TemplateExtensionsPage.tsx

```diff
@@ -0,0 +1,321 @@
+} from '@backstage/core-components';
+import { scaffolderApiRef } from '@backstage/plugin-scaffolder-react';
+import {
+  ScaffolderPageContextMenu,
+  ScaffolderPageContextMenuProps,
+} from '@backstage/plugin-scaffolder-react/alpha';
+import Box from '@material-ui/core/Box';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import ListItemText from '@material-ui/core/ListItemText';
+import Tab from '@material-ui/core/Tab';
+import Tabs from '@material-ui/core/Tabs';
+import TextField from '@material-ui/core/TextField';
+import AllInclusiveIcon from '@material-ui/icons/AllInclusive';
+import FilterListIcon from '@material-ui/icons/FilterList';
+import FunctionsIcon from '@material-ui/icons/Functions';
+import SearchIcon from '@material-ui/icons/Search';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import { trimStart } from 'lodash';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import useAsync from 'react-use/esm/useAsync';
+import {
+  Extension,
+  ExtensionKind,
+  listExtensions,
+  parseLink,
+} from './navigation';
+import { TemplateFilters } from './TemplateFilters';
+import {
+  TemplateGlobalFunctions,
+  TemplateGlobalValues,
+} from './TemplateGlobals';
+
+const useStyles = makeStyles(theme => ({
+  code: {
+    fontFamily: 'Menlo, monospace',
+    padding: theme.spacing(1),
+    backgroundColor:
+      theme.palette.type === 'dark'
+        ? theme.palette.grey[700]
+        : theme.palette.grey[300],
+    display: 'inline-block',
+    borderRadius: 5,
+    border: `1px solid ${theme.palette.grey[500]}`,
+    position: 'relative',
+  },
+
+  codeRequired: {
+    '&::after': {
+      position: 'absolute',
+      content: '"*"',
+      top: 0,
+      right: theme.spacing(0.5),
+      fontWeight: 'bolder',
+      color: theme.palette.error.light,
+    },
+  },
+
+  argRequired: {
+    position: 'relative',
+    '& > *': {
+      display: 'inline',
+      position: 'relative',
+      '&::after': {
+        position: 'absolute',
+        content: '"*"',
+        top: 0,
+        right: theme.spacing(-1),
+        fontWeight: 'bolder',
+        color: theme.palette.error.light,
+      },
+    },
+  },
+
+  link: {
+    paddingLeft: theme.spacing(1),
+    cursor: 'pointer',
+  },
+
+  tabs: {
+    display: 'block',
+    minHeight: 'initial',
+    overflow: 'initial',
+  },
+}));
+
+export const TemplateExtensionsPageContent = ({
+  linkPage,
+}: {
+  linkPage?: string;
+}) => {
+  const api = useApi(scaffolderApiRef);
+  const classes = useStyles();
+  const { t } = useTranslationRef(scaffolderTranslationRef);
+
+  const { loading, value, error } = useAsync(async () => {
+    if (api.listTemplateExtensions) {
+      return api.listTemplateExtensions();
+    }
+    // eslint-disable-next-line no-console
+    console.warn(
+      'listTemplateExtensions is not implemented in the scaffolderApi; please make sure to implement this method.',
```

Comment:
Just to double check, the method exists but is not implemented in the default Scaffolder API implementation, right? I would recommend adding this note to the changeset file as well 🙂

Reply:
> Should we include a note to the effect that custom Scaffolder API impls will need to implement this in order to derive usefulness from this page?

I would say yes, the text in the changeset is the same one people will see in the changelog file. I think it would be frustrating to add the page and have it not work as expected without knowing that this is necessary. But it seems that the default Scaffolder API client already implements this method by default, so maybe the warning is no longer necessary?

Reply:
The reason that things are marked as optional is if adopters are providing their own instance of the `scaffolderApiRef` and ultimately `ScaffolderClient`. 

Marking members and functions as required on the implementation is a breaking change unfortunately, so that's why we mark as optional.

I think it's fine to leave as such and we'll bring everything to required in the next breaking change to the scaffolder.